### DISCUSSION TITLE
[mini.pairs] Pair inside custom pair not closing

### DISCUSSION THREAD

**Discussion by Kayzels** [nvim-mini/mini.nvim#1030] [Q&A]
I've added the following code to my `after/ftplugin/tex.lua` file:
```lua
local mp = require("mini.pairs")
mp.map_buf(0, "i", "$", { action = "closeopen", pair = "$$" })
```

This works perfectly for adding the second $ when I enter math mode in LaTeX by typing the first $. But, any brackets entered inside  the pair don't add the second one, unless there's a space.

Instead of ending up with (where | is the cursor)
```
This is an equation: $f(|)$
```
I get
```
This is an equation: $f(|$
```

How can I get that round bracket to close automatically? It works if I enter a space after the opening $ and before the closing $, but not otherwise. Am I setting it up wrong, or is there something I need to add?

**Comment by echasnovski** [comment]
Unfortunately, I can not reproduce with plain `require('mini.pairs').setup()` in 'init.lua' and snippet you provide placed in 'after/ftplugin/tex.lua'.

It may be that something conflicts with `(` Insert mode mapping. What does `:imap (` show in 'text' buffer? Should be something mentioning 'mini.pairs'.

**Reply by Kayzels** [reply]
Strange that it happens then. I did `:imap (`, and I get
```
v:lua.MiniPairs.open("()", "[^\\].")
Open action for "()" pair
```

I tested with Vimtex disabled, and the same happens. So it's not Vimtex. Seems to happen with all brackets inside it. Maybe because it's right next to the closing bracket or something?

**Reply by Kayzels** [reply]
I tried it with changing the action to be `open` instead of `closeopen`. With that, if I put a space before the closing `$`, then the bracket completion works. But if its right before the closing `$`, then the closing bracket won't be added.

**Reply by echasnovski** [reply]
I am not sure what can cause this then. But I am afraid without reproducible example I can not help.
Here is how it looks for me:


https://github.com/echasnovski/mini.nvim/assets/24854248/9cc95432-0c59-430c-9cc4-9989fc57ed33

**Reply by Kayzels** [reply]
Sorry for spamming, just doing multiple tests. It seems this happens for me with any `closeopen` pair, in any file type.

I'm not sure why this would happen, but suppose I'm in a lua file, and I enter a `"` to open a string. The closing `"` appears, but then if I type a `(` or a `[`, the closing `)` or `]` won't appear.

This happened for me with `"` and `'`, in lua, Python, and a basic text file.

**Reply by echasnovski** [reply]
Still can not reproduce: everything works with any "closeopen" action.
I'd start by disabling everything but 'mini.pairs' and that 'after/ftplugin/tex.lua' and then slowly enabling parts again. That'll help narrow down the cause.

**Comment by Kayzels** [comment]
Thank you, will do. I'm using LazyVim rather than pure Neovim, and it seems that this happens with a default LazyVIm install, so will report it there.
